src/prefect/projects/steps/utility.py

```diff
@@ -0,0 +1,155 @@
+    stderr_sinks = [stderr_sink, sys.stderr] if stream_output else [stderr_sink]
+    async with create_task_group() as tg:
+        tg.start_soon(
+            stream_text,
+            TextReceiveStream(process.stdout),
+            *stdout_sinks,
+        )
+        tg.start_soon(
+            stream_text,
+            TextReceiveStream(process.stderr),
+            *stderr_sinks,
+        )
+
+
+class RunShellScriptResult(TypedDict):
+    """
+    The result of a `run_shell_script` step.
+
+    Attributes:
+        stdout: The captured standard output of the script.
+        stderr: The captured standard error of the script.
+    """
+
+    stdout: str
+    stderr: str
+
+
+async def run_shell_script(
+    script: str,
+    directory: Optional[str] = None,
+    env: Optional[Dict[str, str]] = None,
+    stream_output: bool = True,
+) -> RunShellScriptResult:
+    """
+    Runs a one or more shell commands in a subprocess. Returns the standard
+    output and standard error of the script.
+
+    Args:
+        script: The script to run
+        directory: The directory to run the script in. Defaults to the current
+            working directory.
+        env: A dictionary of environment variables to set for the script
+        stream_output: Whether to stream the output of the script to
+            stdout/stderr
+
+    Returns:
+        A dictionary with the keys `stdout` and `stderr` containing the output
+            of the script
+
+    Examples:
+        Retrieve the short Git commit hash of the current repository to use as
+            a Docker image tag:
+        ```yaml
+        build:
+            - prefect.projects.steps.run_shell_script:
+                id: get-commit-hash
+                script: git rev-parse --short HEAD
+                stream_output: false
+            - prefect.projects.steps.build_docker_image:
+                requires: prefect-docker
+                image_name: my-image
+                image_tag: "{{ get-commit-hash.stdout }}"
+                dockerfile: auto
+        ```
+
+        Run a multi-line shell script:
+        ```yaml
+        build:
+            - prefect.projects.steps.run_shell_script:
+                script: |
+                    echo "Hello"
+                    echo "World"
+        ```
+
+        Run a shell script with environment variables:
+        ```yaml
+        build:
+            - prefect.projects.steps.run_shell_script:
+                script: echo "Hello $NAME"
+                env:
+                    NAME: World
+        ```
+
+        Run a shell script in a specific directory:
+        ```yaml
+        build:
+            - prefect.projects.steps.run_shell_script:
+                script: echo "Hello"
+                directory: /path/to/directory
+        ```
+
+        Run a script stored in a file:
+        ```yaml
+        build:
+            - prefect.projects.steps.run_shell_script:
+                script: "bash path/to/script.sh"
+        ```
+    """
```

Comment:
Would a docs subheading for this make sense? 
(and if you didn't have time, perhaps someone else could take it up)

Reply:
Yeah, making some of these core steps more visible in the docs would be very useful. Do you have an idea of the best place to put examples of steps for visibility?

Reply:
I was thinking here: https://docs.prefect.io/2.10.12/concepts/projects/#the-build-section linked out to a guide called "See examples of running shell scripts in the build step here" or "See examples of build steps here".

Reply:
Unless @billpalombi can think of a better place

Reply:
Its hard to say without seeing the content, but that makes sense to me. I could imagine steps as a top level heading for this page as well.

Reply:
It's just sooo long already hehe

Reply:
@serinamarie added some docs updates in [637f474](https://github.com/PrefectHQ/prefect/pull/9810/commits/637f474ad6ffb6abce42e83170908d8feaacbcea) and [150e1f5](https://github.com/PrefectHQ/prefect/pull/9810/commits/150e1f52c7d1799bd54ae070e4d91a567196293a) based on your suggestions. Let me know what you think!

Reply:
Looks good!